packages/nextjs/src/utils/rewrite.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright © 2024 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+import { OryConfig } from "../types"
+import { joinUrlPaths } from "./utils"
+import { orySdkUrl } from "./sdk"
+
+export function rewriteUrls(
+  source: string,
+  matchBaseUrl: string,
+  selfUrl: string,
+  config: OryConfig,
+) {
+  for (const [_, [matchPath, replaceWith]] of [
+    // TODO load these dynamically from the project config
+    ["/ui/recovery", config.override?.recoveryUiPath],
+    ["/ui/registration", config.override?.registrationUiPath],
+    ["/ui/login", config.override?.loginUiPath],
+    ["/ui/verification", config.override?.verificationUiPath],
+    ["/ui/settings", config.override?.settingsUiPath],
+  ].entries()) {
+    const match = joinUrlPaths(matchBaseUrl, matchPath || "")
+    if (replaceWith && source.startsWith(match)) {
+      source = source.replaceAll(
+        match,
+        new URL(replaceWith, selfUrl).toString(),
+      )
+    }
+  }
+  return source.replaceAll(
+    matchBaseUrl.replace(/\/$/, ""),
+    new URL(selfUrl).toString().replace(/\/$/, ""),
+  )
+}
+export function rewriteJsonResponse<T extends object>(
```

Comment:
I think this needs tests. Seems like a pretty complicated function, maybe some docs/explanations would be good?

Reply:
We have several tests for this in `describe("rewriteJsonResponse", () => {`

I added docs